### PR TITLE
Add handling of dicom references

### PR DESCRIPTION
**Pull request by fedorov**
This topic has been discussed at SLC2015 Project Week, see: http://www.na-mic.org/Wiki/index.php/2015_Winter_Project_Week:DICOM_References

The implementation follows the plan outlined during the project week.

The idea is the following:
1) plugins will populate referencedInstanceUIDs in loadable during Examine
2) on Load, referencedInstanceUIDs corresponding to the selected loadables is
checked, and the files matching those UIDs are located
3) plugins selected (by the user) are invoked to identify loadables corresponding
to the referenced file lists
4) popup dialog with the list of identified loadables (only those 'selected' by
plugins are considered) and the checkboxes allowing the user to cancel loading
5) when the user hits 'Done' button in the popup, selected referenced loadables
are added to the pluginsByLoadable dictionary, and all loadables are loaded per
usual procedure

The proposed patch has been reviewed by @pieper and @cpinter.

### CODE REVIEW

File: Modules/Scripted/DICOMLib/DICOMWidgets.py [hunk 1]
```diff
@@ -505,6 +518,68 @@ def loadCheckedLoadables(self):
       self.examineForLoading()
 
     self.loadableTable.updateSelectedFromCheckstate()
+
+    # TODO: add check that disables all referenced stuff to be considrerd?
```
Comment (jcfr) [inline Modules/Scripted/DICOMLib/DICOMWidgets.py]: Typo in comment

File: Modules/Scripted/DICOMLib/DICOMWidgets.py [hunk 2]
```diff
@@ -505,6 +518,68 @@ def loadCheckedLoadables(self):
...
+    else:
+      self.proceedWithReferencedLoadablesSelection()
+
+    return
+
+  def proceedWithReferencedLoadablesSelection(self):
+
+    # each check box corresponds to a referenced loadable
+    # that was selected by examine; if the user confirmed
+    # that reference should be loaaded, add it to the self.loadablesByPlugin
```
Comment (jcfr) [inline Modules/Scripted/DICOMLib/DICOMWidgets.py]: Typo in comment

**Comment by jcfr** [comment]
Updated topic where I squashed the two commits together has been pushed here: https://github.com/jcfr/Slicer/tree/add-handling-of-dicom-references

**Comment by fedorov** [comment]
Thank you JC! Will you merge it, or is there anything I need to do?

**Comment by jcfr** [comment]
I will integrated it.

On Wed, Jan 28, 2015 at 11:05 PM, Andrey Fedorov <notifications@github.com>
wrote:

> Thank you JC! Will you merge it, or is there anything I need to do?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Slicer/Slicer/pull/231#issuecomment-71965872>.
>



-- 
+1 919 869 8849

**Comment by jcfr** [comment]
Integrated in [r23928](http://viewvc.slicer.org/viewvc.cgi/Slicer4?view=revision&revision=23928)

**Comment by fedorov** [comment]
Thank you!

**Pull request closed without merging**